### PR TITLE
Resolving #4491 , adding support for java maven templates for OpenWhisk

### LINKED ISSUE #4491: Support openwhisk java runtime
# This is a (Feature Proposal)

## Description

Openwhisk supports [java runtime](https://github.com/apache/incubator-openwhisk-runtime-java).
However there is no openwhisk java runtime template in serverless framework.
It's necessary to add openwhisk-java-gradle here and make serverless-openwisk support java runtime
the usecase is like following.

```
serverless create --template openwhisk-java-gradle --path my-service
```

### PR DESCRIPTION
**Pull request by kameshsampath**
Signed-off-by: Kamesh Sampath <kamesh.sampath@hotmail.com>

## What did you implement:

Adding support for creating OpenWhisk Java template with maven as build tool

Closes #4491

## How did you implement it:
Created a another template directory called `openwhisk-java-maven` in `lib/plugins/create/templates/openwhisk-java-maven`, and added 
* Java Sample Function class with required Java Signature of OpenWhisk Java Action
* Java Sample Function Unit Test class
* `serverless.yaml` for OpenWhisk 
* `pom.xml`

## How can we verify it:
Added the UT cases inline with other templates, deploying project successfully to OpenWhisk

## Todos:

- [ ] Write tests
- [*] Write documentation
- [*] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [*] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by horike37** [comment]
Hi, @serverless/ibm 
Could you review this PR?

**Comment by csantanapr** [comment]
@horike37 @jthomas is the maintainer of OpenWhisk plugin
@jthomas  could review this ?

**Comment by kameshsampath** [comment]
@horike37  - fixed the npm issues

@csantanapr @jthomas  - your thoughts? this covers only maven java piece of it, once all finds OK i can push the gradle equivalent as well in similar likes with other providers

File: lib/plugins/create/templates/openwhisk-java-maven/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+service: openwhisk-java-maven # NOTE: update this with your service name
+
+# Please ensure the serverless-openwhisk provider plugin is installed globally.
+# $ npm install -g serverless-openwhisk
+# ...before installing project dependencies to register this provider.
+# $ npm install
+provider:
+  name: openwhisk
+  runtime: java8
```
Comment (jthomas) [inline lib/plugins/create/templates/openwhisk-java-maven/serverless.yml]: This should be `java` or `java:8`?

**Comment by jthomas** [comment]
I've just released Java supported in the provider plugin so this can be merged now. 

I've added a single comment about the `runtime` value, if that is resolved LGTM.

Thanks for submitting this! If you want to add an example using the Java runtimes to https://github.com/serverless/examples/ that would also be great for helping users.

**Comment by csantanapr** [comment]
In openwhisk today the kind  is “java” not java:8 or java8

Once OpenWhisk support another java version then will have
java (for backwards compatibility this will be versión 8)
java:8 (equivalent to java above)
java:9
java:10

**Comment by kameshsampath** [comment]
@csantanapr @jthomas  - I have updated the `serverless.yml` to have java runtime name as `java`. Please review and merge when you are on it.

**Comment by jthomas** [comment]
@horike37 does this now LGTM to you? I'm happy to merge.

**Comment by horike37** [comment]
Thank you for implementing and fixing @kameshsampath ! LGTM :100:

**Pull request merged**